### PR TITLE
mnemosyne: 2.3.2 -> 2.6

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
Mnemosyne was broken previously, because tests failed during install.
I updated it, but still something goes very wrong in the checkPhase if I don't disable checks:
```
running build_ext
Traceback (most recent call last):
  File "nix_run_setup.py", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 178, in <module>
    app = py2app_app
  File "/nix/store/vyn106wj077zhv3n7w9c0nwjwq3xja0z-python3.6-bootstrapped-pip-9.0.1/lib/python3.6/site-packages/setuptools/__init__.py", line 129, in setup
    return distutils.core.setup(**attrs)
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/nix/store/vyn106wj077zhv3n7w9c0nwjwq3xja0z-python3.6-bootstrapped-pip-9.0.1/lib/python3.6/site-packages/setuptools/command/test.py", line 226, in run
    self.run_tests()
  File "/nix/store/vyn106wj077zhv3n7w9c0nwjwq3xja0z-python3.6-bootstrapped-pip-9.0.1/lib/python3.6/site-packages/setuptools/command/test.py", line 248, in run_tests
    exit=False,
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/unittest/main.py", line 94, in __init__
    self.parseArgs(argv)
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/unittest/main.py", line 118, in parseArgs
    self._do_discovery(argv[2:])
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/unittest/main.py", line 229, in _do_discovery
    self.test = loader.discover(self.start, self.pattern, self.top)
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/unittest/loader.py", line 341, in discover
    tests = list(self._find_tests(start_dir, pattern))
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/unittest/loader.py", line 398, in _find_tests
    full_path, pattern, namespace)
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/unittest/loader.py", line 475, in _find_test_path
    tests = self.loadTestsFromModule(package, pattern=pattern)
  File "/nix/store/vyn106wj077zhv3n7w9c0nwjwq3xja0z-python3.6-bootstrapped-pip-9.0.1/lib/python3.6/site-packages/setuptools/command/test.py", line 52, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/unittest/loader.py", line 190, in loadTestsFromName
    return self.loadTestsFromModule(obj)
  File "/nix/store/vyn106wj077zhv3n7w9c0nwjwq3xja0z-python3.6-bootstrapped-pip-9.0.1/lib/python3.6/site-packages/setuptools/command/test.py", line 52, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/unittest/loader.py", line 153, in loadTestsFromName
    module = __import__(module_name)
  File "/tmp/nix-build-mnemosyne-2.6.drv-0/Mnemosyne-2.6/mnemosyne/pyqt_ui/configuration_wdgt_main.py", line 18, in <module>
    Ui_ConfigurationWdgtMain):
  File "/tmp/nix-build-mnemosyne-2.6.drv-0/Mnemosyne-2.6/mnemosyne/pyqt_ui/configuration_wdgt_main.py", line 20, in ConfigurationWdgtMain
    name = _("General")
TypeError: 'NoneType' object is not callable
```

At least, I'm now able to run the program.
Nevertheless, there also seem to be errors within the program. For example, when starting a server and accessing it via web browser (http://localhost:8513/), some error with a `NoneType` occurs.
I sadly don't have the time to look into these and I've also never made contact with PyQt.

/cc @lverns @domenkozar @peti

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by grahamc** [comment]
@GrahamcOfBorg eval

**Comment by dotlambda** [comment]
Is anybody else able to run Mnemosyne from the unstable channel?

**Comment by dotlambda** [comment]
The alternative would be to mark mnemosyne as broken and wait until someone who really needs it comes by. Does this fit you better @FRidh?

**Comment by FRidh** [comment]
Often specifying 
```
checkPhase = ''
  ${python.interpreter} -m unittest discover -s <testsdir>
'';
```
helps with errors like these.

**Comment by FRidh** [comment]
However, if you say the error also occurs during runtime...

You can indeed mark it has broken. Let someone who wants this step up and fix it.

**Comment by dotlambda** [comment]
I think I didn't express myself very well :)

The current status in master is: Mnemosyne builds fine but doesn't run, at least for me:
```
An unexpected error has occurred.
Please forward the following info to the developers:

Traceback (innermost last):
  File "/nix/store/jh87lab7xyxry771ja996sps7j6glwxj-mnemosyne-2.3.2/bin/.mnemosyne-wrapped", line 192, in <module>
    debug_file=options.debug_file)
  File "/nix/store/jh87lab7xyxry771ja996sps7j6glwxj-mnemosyne-2.3.2/lib/python2.7/site-packages/mnemosyne/libmnemosyne/__init__.py", line 174, in initialise
    self.register_components()
  File "/nix/store/jh87lab7xyxry771ja996sps7j6glwxj-mnemosyne-2.3.2/lib/python2.7/site-packages/mnemosyne/libmnemosyne/__init__.py", line 244, in register_components
    exec("from %s import %s" % (module_name, class_name))
  File "<string>", line 1, in <module>
  File "/nix/store/jh87lab7xyxry771ja996sps7j6glwxj-mnemosyne-2.3.2/lib/python2.7/site-packages/mnemosyne/pyqt_ui/qt_web_server.py", line 13, in <module>
    from mnemosyne.web_server.web_server import WebServer
  File "/nix/store/jh87lab7xyxry771ja996sps7j6glwxj-mnemosyne-2.3.2/lib/python2.7/site-packages/mnemosyne/web_server/web_server.py", line 16, in <module>
    from cherrypy import wsgiserver
 ImportError: cannot import name wsgiserver
```

Regarding this PR: Mnemosyne does build AND run. I however wasn't able to make the tests run.

I had another look at the tests and I think we won't be able to enable them:
1. They are not distributed in the tarball.
2. GitHub doesn't provide version tags.
3. GitHub is missing locales etc.
So, we can't use GitHub and therfore we need to diable the tests.

My proposal would be merge this one, since it IS in a working state and I don't see any problem with the disabled tests.

@GrahamcOfBorg build mnemosyne

**Comment by dotlambda** [comment]
ping @FRidh

File: pkgs/games/mnemosyne/default.nix [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
-  name = "mnemosyne-${version}";
+
+buildPythonApplication rec {
+  pname = "mnemosyne";
+  version = "2.6";
+
   src = fetchurl {
-    url    = "http://sourceforge.net/projects/mnemosyne-proj/files/mnemosyne/${name}/Mnemosyne-${version}.tar.gz";
-    sha256 = "0jkrw45i4v24p6xyq94z7rz5948h7f5dspgs5mcdaslnlp2accfp";
+    url    = "http://sourceforge.net/projects/mnemosyne-proj/files/mnemosyne/${pname}-${version}/Mnemosyne-${version}.tar.gz";
```
Comment (FRidh) [inline pkgs/games/mnemosyne/default.nix]: can you use `mirror://sourceforge`?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -18489,7 +18489,9 @@ with pkgs;
     libpng = libpng12;
   };
 
-  mnemosyne = callPackage ../games/mnemosyne { };
+  mnemosyne = libsForQt5.callPackage ../games/mnemosyne {
+    inherit (python3Packages) buildPythonApplication pyqt5 matplotlib cherrypy cheroot webob pillow;
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: Please don't make these kind of changes. They can only cause more work when there are incompatible versions.

File: pkgs/games/mnemosyne/default.nix [hunk 3]
```diff
@@ -1,30 +1,40 @@
...
   preConfigure = ''
     substituteInPlace setup.py --replace /usr $out
     find . -type f -exec grep -H sys.exec_prefix {} ';' | cut -d: -f1 | xargs sed -i s,sys.exec_prefix,\"$out\",
   '';
+
   postInstall = ''
     mkdir -p $out/share
-    mv $out/lib/python2.7/site-packages/$out/share/locale $out/share
-    rm -r $out/lib/python2.7/site-packages/nix
+    mv $out/lib/python*/site-packages/$out/share/locale $out/share
```
Comment (FRidh) [inline pkgs/games/mnemosyne/default.nix]: ${python.sitePackages}

**Comment by dotlambda** [comment]
ping

**Pull request merged**